examples/minimal.rs

```diff
@@ -132,23 +140,34 @@ async fn start_multiple_nodes(
     mut contacts: Vec<SocketAddr>,
     ip: Option<IpAddr>,
     base_port: Option<u16>,
-) {
+) -> Vec<JoinHandle<()>> {
+    let mut handlers = Vec::new();
     let first_index = if first {
-        let first_contact = spawn_first_node(ip, base_port).await;
+        let (first_contact, first_handler): (SocketAddr, JoinHandle<()>) =
+            spawn_first_node(ip, base_port).await;
         contacts.push(first_contact);
+        handlers.push(first_handler);
         1
     } else {
         0
     };
 
     for index in first_index..count {
-        spawn_other_node(index, contacts.clone(), ip, base_port).await;
+        let handler: JoinHandle<()> =
+            spawn_other_node(index, contacts.clone(), ip, base_port).await;
+        handlers.push(handler);
     }
+    handlers
 }
 
 // Spawns the first (genesis) node in its own thread
-async fn spawn_first_node(ip: Option<IpAddr>, base_port: Option<u16>) -> SocketAddr {
-    start_node(0, true, Vec::default(), ip, base_port).await
+async fn spawn_first_node(
```

Comment:
There doesn't seem to be much that these two functions are doing (`spawn_first_node` and `spawn_other_node`). I'd remove them and call `start_node` directly.